### PR TITLE
Test conversions on all SIMD widths

### PR DESCRIPTION
**Pull request by markturner289**
A simple change to testing, most of which was enabled by #548.

Now that the key simulators have been templatised (#548), the last few conversion tests could be moved over to use `TEST_EACH_WORD_SIZE_W` as described in a comment in #559. As an aside, the remaining circuit, cmd and probability tests have also been moved over to be tested across all widths.

### CODE REVIEW

**Review by Strilanc — APPROVED** [review]
Whoops, missed this PR. LGTM after merge conflicts resolved.

**Comment by markturner289** [comment]
Great, merge conflicts fixed.

**Pull request merged**
